cache/sharded_cache.h

```diff
@@ -111,19 +108,29 @@ class ShardedCache : public Cache {
                                DeleterFn deleter)>& callback,
       const ApplyToAllEntriesOptions& opts) override;
   virtual void EraseUnRefEntries() override;
-  virtual std::string GetPrintableOptions() const override;
 
   int GetNumShardBits() const;
   uint32_t GetNumShards() const;
 
+  Status ValidateOptions(const DBOptions& db_opts,
+                         const ColumnFamilyOptions& cf_opts) const override;
+
+  bool IsInstanceOf(const std::string& id) const override {
+    if (id == kClassName()) {
+      return true;
+    } else {
+      return Cache::IsInstanceOf(id);
+    }
+  }
+
  protected:
+  uint32_t SetNumShards(int num_shard_bits);
+
   inline uint32_t Shard(uint32_t hash) { return hash & shard_mask_; }
+  mutable port::Mutex capacity_mutex_;
```

Comment:
I would call this `options_mutex_` and comment that it is used to ensure consistency between the "source of truth" `options_` in each derived class and the downstream option settings in each cache shard. This "source of truth" is not normally read during performance-critical operations, so a Cache-wide mutex is OK.

Reply:
Previously it was only used for capacity-related mutability. Now it's clearly generalized to guarding all dynamic options, based on new usage in `IsMutable` and `PrepareOptions`

Reply:
Done.